tests/console/zypper_ar.pm

```diff
@@ -25,7 +25,12 @@ sub run {
         foreach (@repos_to_add) {
             next unless get_var("REPO_$_");    # Skip repo if not defined
             $repourl = $urlprefix . "/" . get_var("REPO_$_");
-            assert_script_run "zypper lr | grep -w $_ || zypper ar -c $repourl $_";    # Skip add repo if already added
+            # Skip add repo if already added
+            my $rc = script_run "zypper lr | grep -w $_ || zypper ar -c $repourl $_";
+            # treat OSS error as test failure, others can be just recorded
+            if ($rc) {
+                ($_ =~ m/^\[OSS/) ? die 'OSS repo failed!' : record_info("$_ repo failure", "zypper exited with code $rc");
```

Comment:
I wonder why do we need `[` in the regexp? And just a minor, what's about writing 'Adding OSS repo failed?' I guess it's a bit more clear then. WDYT?

Reply:
That makes more sense. I will update it